torch/csrc/jit/tensorexpr/loopnest.cpp

```diff
@@ -751,28 +777,50 @@ bool LoopNest::computeInline(const Buf* b) {
   return true;
 }
 
-void LoopNest::inlineIntermediateBufs(bool inline_output_buffers) {
+// inlining buffers with multiple uses creates duplicated work, which can slow
+// down cpu code generation but is enabled on gpu because it avoids difficult
+// synchronization logic across blocks.
+void LoopNest::inlineIntermediateBufs(bool allow_duplicated_work) {
   // We need to collect all intermediate buffers as the buffers to be inlined
   // before calling 'computeInline' since the buffers that are inlined are
   // erased from the set 'intermediate_bufs_' in that function.
-  std::unordered_set<const Buf*> bufs_to_inline(
-      intermediate_bufs_.begin(), intermediate_bufs_.end());
+  std::unordered_set<const Buf*> bufs_to_inline;
 
-  // inlining output buffers duplicates computation. it slows down
-  // cpu code generation but is enabled on gpu because it avoids difficult
-  // synchronization logic across blocks.
-  if (inline_output_buffers) {
+  if (allow_duplicated_work) {
```

Comment:
Nit: might be worth mentioning that inlining a statement with multiple uses doesn't *always* results in a duplicated work: trivial copies (a[i] = b[i]) can be inlined everywhere and it will not be more expensive than the non-inlined version.

Reply:
about that... see comment, what do you think?